src/wallet/wallet.h

```diff
@@ -475,7 +475,38 @@ class CWalletTx : public CMerkleTx
 };
 
 
-using CInputCoin = std::pair<const CWalletTx*, unsigned int>;
+class CInputCoin {
+public:
+    CInputCoin()

```

Comment:
Would suggest deleting default constructor and IsNull method. Their presence seem to make this class more complex and precarious than is necessary, and could lead to bugs and unintended behavior.

The only place where a nullable `CInputCoin` seems to be used is in one `coinLowestLarger` local variable, and you could more easily declare this variable as `boost::optional<CInputCoin> coinLowestLarger;` (or [std::optional](http://en.cppreference.com/w/cpp/utility/optional) in the future) instead of building in a layer of optionality into the type.